uniffi/src/bindings/kotlin/templates/Helpers.kt

```diff
@@ -0,0 +1,28 @@
+// A handful of classes and functions to support the generated data structures.
+// This would be a good candidate for isolating in its own ffi-support lib.
+
+abstract class FFIObject {
+    internal val isDestroyed = AtomicBoolean(false)
```

Comment:
Something we've done in the hand-written bindings, was use `this.handle == 0` to indicate that the object is dead. Would it make sense to move `handle` up to a property of this parent class in order to do something like that?

Reply:
I'm not overly happy with this object being a class— having it as a `AtomicBoolean` would allow this to become an interface again.